tests/test_anonfile.py

```diff
@@ -38,73 +39,116 @@ def remove_file(file: str) -> None:
     Path(file).unlink() if Path(file).exists() else None
 
 
-class TestAnonFileLibrary(unittest.TestCase):
+class MockData:
+    @staticmethod
+    def get_json_response(filename):
+        with open(filename, "r", encoding="utf-8") as json_file:
+            json_response = Mock(spec=Response)
+            json_response.__enter__ = MagicMock(return_value=json_response)
+            json_response.__exit__ = MagicMock()
+            json_response.status_code = 200
+            json_response.json.return_value = json.loads(json_file.read())
+            return json_response
+
+    @staticmethod
+    def get_html_response(filename):
+        with open(filename, "r", encoding="utf-8") as html_file:
+            html_response = Mock(spec=Response)
+            html_response.__enter__ = MagicMock(return_value=html_response)
+            html_response.__exit__ = MagicMock()
+            html_response.status_code = 200
+            html_response.text = html_file.read()
+            return html_response
+
+    @staticmethod
+    def get_file_response(filename):
+        with open(filename, "rb") as file:
+            file_response = Mock(spec=Response)
+            file_response.__enter__ = MagicMock(return_value=file_response)
+            file_response.__exit__ = MagicMock()
+            file_response.status_code = 200
+            file_response.iter_content.return_value = (chunk for chunk in file.read())
+            return file_response
+
+
+class TestAnonFile(unittest.TestCase):
+    """Test cases for the AnonFile class."""
+
     @classmethod
     def setUpClass(cls):
         cls.anon = init_anon()
-        cls.test_file = Path("tests/test.txt")
+        cls.test_file = Path("tests/original_topsecret.mp4")
         cls.test_small_file = "https://anonfiles.com/93k5x1ucu0/test_txt"
-        cls.test_med_file = "https://anonfiles.com/Z6j7c9V6x5/topsecret_mp4"
+        cls.test_med_file = "https://anonfiles.com/P0mev3tfz7/topsecret_mp4"
         cls.garbage = []
 
-    def test_upload(self):
+    @patch('anonfile.requests.Session.post')
+    def test_upload(self, mocked_session_post):
+        """ Tests a mocked upload """
+
+        # Given
+        json_response = MockData.get_json_response("tests/upload.json")
+
+        mocked_session_post.side_effect = [json_response]
+
+        # Assert
         upload = self.anon.upload(self.test_file, progressbar=True, enable_logging=True)
         self.assertTrue(upload.status, msg="Expected 200 HTTP Error Code")
         self.assertTrue(all([upload.url.scheme, upload.url.netloc, upload.url.path]), msg="Invalid URL.")
 
-    def test_preview(self):
-        preview = self.anon.preview(self.test_small_file)
+    @patch('anonfile.requests.Session.get')
+    def test_preview(self, mocked_session_get):
+        """ Tests mocked preview API request """
+
+        # Given
```

Comment:
Remember your AAAs:

1. Arrange
2. Act
3. Assert

This is a widely used to convention to check if a test is complete. It's also the first thing that I write down when I implement a new test :)

Reply:
This was solid advice, and turned me onto some very useful information.